src/java.base/linux/classes/jdk/internal/platform/cgroupv1/CgroupV1Subsystem.java

```diff
@@ -0,0 +1,490 @@
+                case "cpuacct":
+                    subsystem.setCpuAcctController(new CgroupV1SubsystemController(mountentry[3], mountentry[4]));
+                    break;
+                case "cpu":
+                    subsystem.setCpuController(new CgroupV1SubsystemController(mountentry[3], mountentry[4]));
+                    break;
+                case "blkio":
+                    subsystem.setBlkIOController(new CgroupV1SubsystemController(mountentry[3], mountentry[4]));
+                    break;
+                default:
+                    // Ignore subsystems that we don't support
+                    break;
+            }
+        }
+    }
+
+    /**
+     * setSubSystemPath based on the contents of /proc/self/cgroup
+     */
+    private static void setSubSystemControllerPath(CgroupV1Subsystem subsystem, String[] entry) {
+        String controllerName;
+        String base;
+        CgroupV1SubsystemController controller = null;
+        CgroupV1SubsystemController controller2 = null;
+
+        controllerName = entry[1];
+        base = entry[2];
+        if (controllerName != null && base != null) {
+            switch (controllerName) {
+                case "memory":
+                    controller = subsystem.memoryController();
+                    break;
+                case "cpuset":
+                    controller = subsystem.cpuSetController();
+                    break;
+                case "cpu,cpuacct":
+                case "cpuacct,cpu":
+                    controller = subsystem.cpuController();
+                    controller2 = subsystem.cpuAcctController();
+                    break;
+                case "cpuacct":
+                    controller = subsystem.cpuAcctController();
+                    break;
+                case "cpu":
+                    controller = subsystem.cpuController();
+                    break;
+                case "blkio":
+                    controller = subsystem.blkIOController();
+                    break;
+                // Ignore subsystems that we don't support
+                default:
+                    break;
+            }
+        }
+
+        if (controller != null) {
+            controller.setPath(base);
+            if (controller instanceof CgroupV1MemorySubSystemController) {
+                CgroupV1MemorySubSystemController memorySubSystem = (CgroupV1MemorySubSystemController)controller;
+                boolean isHierarchial = getHierarchical(memorySubSystem);
+                memorySubSystem.setHierarchical(isHierarchial);
+            }
+            subsystem.setActiveSubSystems();
+        }
+        if (controller2 != null) {
+            controller2.setPath(base);
+        }
+    }
+
+
+    private static boolean getHierarchical(CgroupV1MemorySubSystemController controller) {
+        long hierarchical = getLongValue(controller, "memory.use_hierarchy");
+        return hierarchical > 0;
+    }
+
+    private void setActiveSubSystems() {
```

Comment:
function `getSwapEnabled()` from JDK-8250984 is missing here. See:
https://github.com/openjdk/jdk11u-dev/blob/194b66f6a72437345c458e747e24beb46352537a/src/java.base/linux/classes/jdk/internal/platform/cgroupv1/Metrics.java#L246..L249